File: 1843-number-of-rectangles-that-can-form-the-largest-square/number-of-rectangles-that-can-form-the-largest-square.py
class Solution(object):
    def countGoodRectangles(self, rectangles):
        """
        :type rectangles: List[List[int]]
        :rtype: int
        """
        max_side_lengths = [min(l, w) for l, w in rectangles]
        
        max_len = max(max_side_lengths)
        
        count = sum(1 for side in max_side_lengths if side == max_len)
        
        return count

File: 1843-number-of-rectangles-that-can-form-the-largest-square/number-of-rectangles-that-can-form-the-largest-square.js
/**
 * @param {number[][]} rectangles
 * @return {number}
 */
var countGoodRectangles = function(rectangles) {
    let lengths = [], maxLen = 0, count = 0;

    for (let length of rectangles) {
        let l = length[0], w = length[1], k = Math.min(l, w);
        lengths.push(k);
        if (maxLen < k) {
            maxLen = k;
        }
    }

    lengths.forEach((element) => {
        if(element === maxLen){
            count += 1;
        }
    })

    return count;
};